src/rust/engine/src/session.rs

```diff
@@ -139,6 +139,12 @@ pub struct Session {
     state: Arc<SessionState>,
 }
 
+#[derive(Clone, Debug)]
+pub struct WeakSession {
```

Comment:
This is needed because the session has a circular reference to the GitignoreSession?